tests/sccache_cargo.rs

```diff
@@ -133,5 +142,34 @@ fn test_rust_cargo_cmd(cmd: &str, extra_envs: &[(&str, &std::ffi::OsStr)]) {
         .assert()
         .stdout(predicates::str::contains(r#""cache_hits":{"counts":{"Rust":2}}"#).from_utf8())
         .success();
+
+    if cmd == "run" {
```

Comment:
The test for the env vars is now separate. While doing so I refactored basically the whole test and I hope it is easier to add new tests now without duplicating a lot of code.

I removed the deny for rust 2018 idioms from the integration test file, since the test itself should always be built with a cargo version >= the sccache MSRV (which I guessed to be 1.48 by looking at CI). 

Depending on how important the `cfg` excluding windows and macos was, I may also just put everything in the test into a mod.